ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -138,12 +134,6 @@ object LedgerClient {
       builder: NettyChannelBuilder,
       configuration: LedgerClientConfiguration,
   )(implicit ec: ExecutionContext, esf: ExecutionSequencerFactory): Future[LedgerClient] = {
-    val channel = GrpcChannel(builder, configuration)
-    sys.addShutdownHook {
-      channel.shutdownNow()
-      ()
-    }
-    apply(channel, configuration)
+    apply(GrpcChannel.withShutdownHook(builder, configuration), configuration)
```

Comment:
```suggestion
    LedgerClient(GrpcChannel.withShutdownHook(builder, configuration), configuration)
```
Not necessary to change this, the argument I would make is that this eliminates any ambiguity for the reader as to which `apply` method is this.

Reply:
I agree, thanks!